src/geometry/types.rs

```diff
@@ -5,15 +5,20 @@ use std::cmp::PartialEq;
 //
 // Coordinates in 2D space.
 
-#[derive(PartialEq, Debug)]
+#[derive(Clone, PartialEq, Debug)]
```

Comment:
so we can use clone() function without defining it ?

Reply:
Yes, you can derive `Clone`. If you derive it, then it makes a copy by calling `clone()` on all of the properties.